app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt

```diff
@@ -45,6 +44,25 @@ class TakePhotoActivity : AppCompatActivity() {
         }
     }
 
+    private fun hasPermission(): Boolean {
+        val reqPermission = ContextCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED
```

Comment:
can you rename `reqPermission` to something clearer that shows it's a Boolean? like  `shouldRequestPermission`